tests/rptest/tests/topic_recovery_test.py

```diff
@@ -1080,10 +1086,13 @@ def do_run(self, test_case: BaseCase):
         test_case.generate_baseline()
 
         # Give time to finish all uploads
-        time.sleep(10)
+        self.logger.info(f"Waiting {upload_delay_sec} sec for S3 uploads...")
+        time.sleep(upload_delay_sec)
```

Comment:
I feel like we should wait on a condition here instead of a fixed time.

Reply:
I generally agree. This is an existing test so I left it for now. One good thing about a sleep here is it saves us some $$ on S3 metadata queries. The bigger problem with this code is that it doesn't seem to reliably find all the segments it expects, even with very long (over an hour) wait times. This is with this scale test though--which creates a ton of segments.